src/Patterns/Path.php

```diff
@@ -1,413 +1,145 @@
+    use ErrorTrait;
 
-    /**
-     * @var StructuralType The pattern right of the relationship
-     */
-    private StructuralType $right;
-
-    /**
-     * @var string[] The direction of the relationship
-     */
-    private array $direction;
-
-    /**
-     * @var string[]
-     */
-    private array $types = [];
-
-    /**
-     * @var Variable|null
-     */
-    private ?Variable $variable = null;
-
-    /**
-     * @var int|null The minimum number of `relationship->node` hops away to search
-     */
-    private ?int $minHops = null;
+    /** @var Relationship[] */
+    private array $relationships;
+    /** @var Node[] */
+    private array $nodes;
 
     /**
-     * @var int|null The maximum number of `relationship->node` hops away to search
+     * @param AnyType|AnyType[]|null $nodes
+     * @param Relationship|Relationship[]|null $relationships
      */
-    private ?int $maxHops = null;
-
-    /**
-     * @var int|null The exact number of `relationship->node` hops away to search
-     */
-    private ?int $exactHops = null;
-
-    /**
-     * @var MapType|null
-     */
-    private ?MapType $properties = null;
-
-    /**
-     * Path constructor.
-     *
-     * @param StructuralType $left The node left of the relationship
-     * @param StructuralType $right The node right of the relationship
-     * @param array $direction The direction of the relationship, should be either:
-     *                           - Path::DIR_RIGHT (for a relation of
-     *                           (a)-->(b)) - Path::DIR_LEFT (for a relation
-     *                           of (a)<--(b)) - Path::DIR_UNI (for a
-     *                           relation of (a)--(b))
-     */
-    public function __construct(StructuralType $left, StructuralType $right, array $direction)
+    public function __construct($nodes = null, $relationships = null)
     {
-        $this->left = $left;
-        $this->right = $right;
+        self::assertClass('relationships', [Relationship::class, 'array', 'null'], $relationships);
+        self::assertClass('nodes', [AnyType::class, 'array', 'null'], $nodes);
 
-        if ($direction !== self::DIR_RIGHT && $direction !== self::DIR_LEFT && $direction !== self::DIR_UNI) {
-            throw new InvalidArgumentException("The direction must be either 'DIR_LEFT', 'DIR_RIGHT' or 'RELATED_TO'");
-        }
+        $nodes ??= [];
+        $relationships ??= [];
 
-        $this->direction = $direction;
+        $this->nodes = is_array($nodes) ? array_values($nodes) : [$nodes];
+        $this->relationships = is_array($relationships) ? array_values($relationships) : [$relationships];
     }
 
-    /**
-     * Add the given property to the properties of this path.
-     *
-     * @param string $key The name of the property
-     * @param AnyType $value The value of the property
-     * @return PathType
-     */
-    public function withProperty(string $key, AnyType $value): PathType
+    public function toQuery(): string
```

Comment:
Some doc comments explaining what each function does would be helpful in understanding this class.

Reply:
I added comprehensive documentation on each step now.